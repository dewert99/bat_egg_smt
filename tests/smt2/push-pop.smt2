(declare-sort S)
(declare-const a S)
(declare-const b S)
(declare-fun f (S) Bool)
(assert (= a b))
(assert (f a))
(check-sat)
(get-model)
(push)
(declare-sort S2)
(declare-fun g (S2) S)
(declare-const c S2)
(assert (= b (g c)))
(check-sat)
(get-model)
(assert (not (f (g c))))
(check-sat)
(get-unsat-core)
(pop)
(declare-const c S)
(declare-sort S2 3)
(assert (not (f c)))
(check-sat)
(get-model)
(assert (= b c))
(check-sat)
(get-unsat-core)