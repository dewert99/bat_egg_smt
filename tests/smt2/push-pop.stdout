sat
(
 (define-fun a () S @S_3)
 (define-fun b () S @S_3)
 (define-fun f ((x!0 S)) Bool
  (ite (= x!0 @S_3) true
   f!default))
)
sat
(
 (define-fun a () S @S_3)
 (define-fun b () S @S_3)
 (define-fun c () S2 @S2_6)
 (define-fun f ((x!0 S)) Bool
  (ite (= x!0 @S_3) true
   f!default))
 (define-fun g ((x!0 S2)) S
  (ite (= x!0 @S2_6) @S_7
   g!default))
)
unsat
()
sat
(
 (define-fun a () S @S_3)
 (define-fun b () S @S_3)
 (define-fun c () S @S_6)
 (define-fun f ((x!0 S)) Bool
  (ite (= x!0 @S_3) true
  (ite (= x!0 @S_6) false
   f!default)))
)
unsat
()
